openapi/src/main/scala/endpoints/openapi/DocumentedBasicAuthentication.scala

```diff
@@ -0,0 +1,20 @@
+package endpoints
+package openapi
+
+import endpoints.algebra.BasicAuthentication.Credentials
+
+/**
+  * Interpreter for [[algebra.BasicAuthentication]] that produces
+  * OpenAPI documentation.
+  */
+trait DocumentedBasicAuthentication
+  extends algebra.DocumentedBasicAuthentication
+    with DocumentedEndpoints {
+
+  private[endpoints] def basicAuthentication: RequestHeaders[Credentials] =
```

Comment:
Have i missed this before or is this something new? I mean limiting access in interpreters. I find it very nice, because namespace of client/server is not cluttered.